### PR TITLE
fix: delete table when source removed

### LINKED ISSUE #309: CSV Mapping: Deleting a Input CSV doesn't delete its Output tables
Steps to reproduce:
- Create a new project
- add a CSV
- Create a table "Tabelle1"
- Delete the Output table "Tabelle1" (proof for correct behaviour)
- Create a table "Tabelle2"
- Try "Link to another table" 
  - --> OK, the "Link to table"-Dropdown will only contain "Tabelle2"
- Delete the whole Input CSV
  - --> OK, the associated output table disappears too
- Upload another CSV
- Create a table "Tabelle3"
- Try "Link to another table" 
  - --> BUG, the "Link to table"-Dropdown will contain "Tabelle2" and  "Tabelle3".  "Tabelle2" should've been deleted.

Remarks
- Having the table itself on the list would create a recursion, I don't think this should be allowed (cf. #310)

### PR DESCRIPTION
**Pull request by tpluscode**
@martinmaillard @lucafurrer do you remember why we only unset the table property and not actually remove the table?

fixes #309

### CODE REVIEW

**Comment by martinmaillard** [comment]
Not really. Probably because "maybe in the future we will X". I also think it's safer to delete the table.

**Comment by lucafurrer** [comment]
If I remember right, there was the idea, that you can delete the source upload a new one and reconnect it with the new source. The idea at the time was that the table is not strongly linked to the source.

I could look through my discussions etc, but I think it is not worth the time.

File: apis/core/test/domain/csv-source/delete.test.ts [hunk 1]
```diff
@@ -25,17 +22,16 @@ describe('domain/csv-sources/delete', () => {
...
 
-    getLinkedTablesForSource = sinon.stub().returns(tableGenerator())
-    getTablesForMapping = sinon.stub().returns(tableGenerator())
-    tableQueries = {
-      getLinkedTablesForSource,
-      getTablesForMapping,
-      getTableForColumnMapping,
-    }
+    sinon.restore()
+    sinon.stub(TableQueries, 'getLinkedTablesForSource').returns(tableGenerator())
```
Comment (martinmaillard) [inline apis/core/test/domain/csv-source/delete.test.ts:30]: What made you decide to stub stuff instead of injecting dependencies?

**Review by martinmaillard — APPROVED** [review]

File: apis/core/test/domain/csv-source/delete.test.ts (hunk 1 above)
Comment (tpluscode) [reply apis/core/test/domain/csv-source/delete.test.ts:30]: I think it's cleaner on the code which actually uses them. What do you think?

Comment (martinmaillard) [reply apis/core/test/domain/csv-source/delete.test.ts:30]: I'm fine with both. Mocking in this way makes the code a bit simpler, but the test a bit more brittle. The tradeoff is probably worth it.

**Pull request merged**